src/main/java/org/terasology/taskSystem/actions/PerformTaskNode.java

```diff
@@ -122,4 +160,42 @@ private void changeOreonAttributes(Actor oreon, TaskComponent taskComponent) {
 
         oreon.save(oreonAttributeComponent);
     }
+
+    /**
+     * Places the required blocks in the selected area based on the task selected
+     * @param oreon The Oreon entity working on the task
+     * @param taskComponent The component with task information which just completed
+     */
+    private void completeTask(Actor oreon, TaskComponent taskComponent) {
+        Region3i selectedRegion = taskComponent.taskRegion;
+        String taskType = taskComponent.assignedTaskType;
+
+
+        switch (taskType) {
+            case AssignedTaskType.Plant :
+                plantingTaskCompletion.placeCrops(selectedRegion, Constants.OREON_CROP_0_BLOCK);
+                break;
+
+            case AssignedTaskType.Build :
+                constructingFromStructureTemplate.constructBuilding(selectedRegion, taskComponent.buildingType);
+                //constructingFromBuildingGenerator.constructBuilding(selectedRegion, taskComponent.buildingType);
+                break;
+
+            case AssignedTaskType.Upgrade :
+                oreon.getEntity().send(new BuildingUpgradeStartEvent());
+        }
+    }
+
+    private void setPlantingTaskCompletion() {
+        this.plantingTaskCompletion = new PlantingTaskCompletion(blockManager, blockEntityRegistry);
+    }
+
+    private void setConstructingFromStructureTemplate(Actor oreon) {
+        EntityRef player = oreon.getComponent(OreonSpawnComponent.class).parent;
+        this.constructingFromStructureTemplate = new ConstructingFromStructureTemplate(structureTemplateProvider, player);
+    }
+
+    private void setConstructingFromBuildingGenerator() {
```

Comment:
Don't see why this is in a separate private method.

Reply:
I am implemented these as separate methods to keep the `construct` method a bit clean.

Reply:
They don't really make the code cleaner if they are one-liners replaced by a one-liner method call.